packages/dds/tree/src/test/shared-tree/summaryBenchmarkSizes.ts

```diff
@@ -0,0 +1,173 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import assert from "assert";
+import { IsoBuffer } from "@fluidframework/common-utils";
+import { makeRandom } from "@fluid-internal/stochastic-test-utils";
+import { TransactionResult } from "../../checkout";
+import { FieldKinds, singleTextCursor } from "../../feature-libraries";
+import { ISharedTree } from "../../shared-tree";
+import { FieldKey, rootFieldKey, rootFieldKeySymbol, TreeValue } from "../../tree";
+import { brand } from "../../util";
+import { TestTreeProvider } from "../utils";
+import { fieldSchema, GlobalFieldKey, namedTreeSchema, SchemaData } from "../../schema-stored";
+// eslint-disable-next-line import/no-internal-modules
+import { PlacePath } from "../../feature-libraries/sequence-change-family";
+
+const globalFieldKey: GlobalFieldKey = brand("globalFieldKey");
+
+describe("Summary size benchmark", () => {
+    it("with no nodes.", async () => {
+        const provider = await TestTreeProvider.create(1);
+        const tree = provider.trees[0];
+
+        const { summary } = tree.getAttachSummary();
+        const summaryString = JSON.stringify(summary);
+        const summarySize = IsoBuffer.from(summaryString).byteLength;
+        assert(summarySize < 1000);
+    });
+    it("with 1 inserted node.", async () => {
+        const summarySize = await getInsertsSummarySize(1, 0, false);
+        assert(summarySize < 2000);
+    });
+    it("with 10 inserted nodes width-wise.", async () => {
+        const summarySize = await getInsertsSummarySize(10, 0, false);
+        assert(summarySize < 3000);
+    });
+    it("with 100 inserted nodes width-wise.", async () => {
+        const summarySize = await getInsertsSummarySize(100, 0, false);
+        assert(summarySize < 20000);
+    });
+    it("with 10 inserted nodes depth-wise.", async () => {
+        const summarySize = await getInsertsSummarySize(10, 0, true);
+        assert(summarySize < 3000);
+    });
+    it("with 100 inserted nodes depth-wise.", async () => {
+        const summarySize = await getInsertsSummarySize(100, 0, true);
+        assert(summarySize < 20000);
+    });
+    it("rejected for 1000 inserts depth wise", async () => {
+        await assert.rejects(getInsertsSummarySize(1000, 0, true), { message: "BatchTooLarge" });
```

Comment:
this seems like a fluid framework bug, not a feature of summary sizes. At the very least we should document that this should not happen instead of making it look like this test is ensuring that it does happen (ex: this test could prevent someone from fixing fluid to handle larger ops because it breaks this test).


```suggestion
        // TODO: this should work, but currently hits batch size limits. Convert this into benchmark when fixed.
        await assert.rejects(getInsertsSummarySize(1000, 0, true), { message: "BatchTooLarge" });
```

Reply:
added TODO comment